health_monitor/src/health_monitor.cpp

```diff
@@ -139,8 +139,121 @@ namespace health_monitor
         ros::Duration sd(startup_duration_);
         long start_duration=sd.toNSec() / 1e6;
 
-        nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+        
+        /*Publish each system alert only once*/
+        auto crit_driver_status_truck = driver_manager_.are_critical_drivers_operational_truck(time_now);
+        auto crit_driver_status_car = driver_manager_.are_critical_drivers_operational_car(time_now);
+
+        if(truck_ == true)
+        {
+            if((crit_driver_status_truck == "s_1_l1_1_l2_1_g_1") && (is_published_truck[0] == false)) //"All essential drivers are ready"
+            {
+             	nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
```

Comment:
Seems like you could call driver_manager_.handleSpin once before the conditionals and cache the return values. It would clean up the contents of these publish calls quite a bit.